testlapack/dsterf.go

```diff
@@ -0,0 +1,124 @@
+// Copyright ©2016 The gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package testlapack
+
+import (
+	"math"
+	"math/rand"
+	"sort"
+	"testing"
+
+	"github.com/gonum/floats"
+)
+
+type Dsterfer interface {
+	Dgetrfer
+	Dsterf(n int, d, e []float64) (ok bool)
+}
+
+func DsterfTest(t *testing.T, impl Dsterfer) {
+	// Hand coded tests
+	for cas, test := range []struct {
+		d []float64
+		e []float64
+		n int
+
+		ans []float64
+	}{
+		{
+			d:   []float64{1, 3, 4, 6},
```

Comment:
I was thinking more of the origin of the ans than the origin of the input - unless you have an internal implementation of dsterf.